### PR TITLE
slim_model: options for float16, checkpoint selection

### PR DESCRIPTION
**Pull request by nostalgebraist**
Contains some additions I made to `slim_model.py` when I was using it with a finetuned model.

- `--f16`: converts to float16
  - Why: Since many people are doing inference with float16 weights, it seems better to just go directly to float16, without the bfloat16 step

- `--ckpt-step`: passing e.g. `--ckpt-step 401` will load from `step_401/` even if it's not the most recent checkpoint
  - Why: I wanted to convert the model with the lowest val loss, but usually there were 1 or more checkpoints after that step

### CODE REVIEW

File: slim_model.py [hunk 1]
```diff
@@ -11,13 +11,15 @@
 from mesh_transformer.transformer_shard import CausalTransformer
 from smart_open import open
 
-from mesh_transformer.util import clip_by_global_norm, to_bf16
+from mesh_transformer.util import clip_by_global_norm, to_bf16, to_f16
```
Comment (kingoflolz) [inline slim_model.py:14]: I think you forgot to include your to_f16 function

**Review by kingoflolz — CHANGES_REQUESTED** [review]
Otherwise looks good!

File: slim_model.py (hunk 1 above)
Comment (nostalgebraist) [reply slim_model.py:14]: Oh whoops, I thought it was already in the repo... pushed

**Comment by kingoflolz** [comment]
Thanks!
